### PR TITLE
Enable EE9 run in the JSP 2.3 FAT

### LINKED ISSUE #11371: JakartaEE: Enable EE9 RepeatTests in JSP 2.3 FAT
Enable JSP FATs to run against the JakartaEE transformed JSP feature.

See epic #9876.

### LINKED ISSUE #11612: JakartaEE: Enable EE9 RepeatTests in JSF 2.2/2.3 FAT
Enable JSF FATs to run against the JakartaEE transformed JSF feature.

### LINKED ISSUE #11732: @ExpectedFFDC to handle Javax & Jakarta Exceptions
Current FatRunner is not designed to handle exceptions from 2 different namespaces.

### LINKED ISSUE #11825: JakartaEE: Update EL-4.0 API to use Tomcat EL
When EL-4.0 was created, I had used the API from the Eclipse EE4J Project.  However, a few test failures occur when used with jasper's implementation.  While we coudl update the implementation, Jasper is integrated with IBM JSP, and would require large changes to update to EE4J version.  For this reason, it would better to stay with Tomcat. Additionally, it would also keep it consistent with EL-3.0

### PR DESCRIPTION
**Pull request by volosied**
Fixes #11732  
Fixes #11371 
Fixes #11825 

Several tests rely on CDI, so not all of them can run under EE9 yet (waiting on cdi-3.0).  Since they were all in the TestJSP2.3.war, it caused the bucket to be very slow due to the mix of javax/jakarta. I refactored the tests around, so the CDI tests are now in the JSPCdiTest class (and can be skip the EE9 run).  I also changed the test applications around to better reflect what they test.

Additionally, I updated the FATRunner to transform the AllowedFFDC/ExpectedFFDC annotations automatically based on the jakarta rules files. 

There were a few EL Failures ( see below) when using the EE4J EL API. I believe it is related to API/IMPL differences (since the implementation is from Tomcat). I've changed the API to Tomcat in EL 4.0, and the issues seemed to have gone away. The differences shown in git reflect the changes between the Jakarta/Tomcat APIs.

testJSP23ResolutionVariableProperties_EE9_FEATURES
 testEL30LambdaExpressions_EE9_FEATURE

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java [hunk 1]
```diff
@@ -336,9 +330,8 @@ private void nonReservedWordsHelperMethod(WebConversation wc, String path, Strin
      *                       if something goes horribly wrong
      */
     @Test
-    @SkipForRepeat("CDI-2.0") // No need to run against cdi-2.0 since these tests don't use CDI at all.
     public void testServlet() throws Exception {
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java:333]: Why were these SkipForRepeat annotations removed in this class?

File: dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspCdiServer/bootstrap.properties [hunk 2]
```diff
@@ -0,0 +1,14 @@
+###############################################################################
+# Copyright (c) 2014 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspCdiServer/bootstrap.properties:3]: For a new file this copyright seems strange

File: dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspCdiServer/server.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<!--
+    Copyright (c) 2015 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspCdiServer/server.xml:3]: For a new file this copyright seems strange

File: dev/com.ibm.ws.jsp.2.3_fat/test-applications/TestServlet.war/resources/WEB-INF/web.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        IBM Corporation - initial API and implementation
+ -->
+<web-app
+    version="3.1"
+    xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd">
+
+  <display-name>TestJSP2.3</display-name>
+
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/test-applications/TestServlet.war/resources/WEB-INF/web.xml:19]: If we're going to change the app name we should update the display-name as well for any apps we changed the names of.

File: dev/fattest.simplicity/src/componenttest/custom/junit/runner/FATRunner.java [hunk 5]
```diff
@@ -56,6 +56,7 @@
 import componenttest.exception.TopologyException;
 import componenttest.logging.ffdc.IgnoredFFDCs;
 import componenttest.logging.ffdc.IgnoredFFDCs.IgnoredFFDC;
+import componenttest.rules.repeater.EE9PackageReplacementHelper;
```
Comment (pnicolucci) [inline dev/fattest.simplicity/src/componenttest/custom/junit/runner/FATRunner.java:59]: I'd suggest updating the copyright date range for this file. Looks like pretty significant functional changes.

File: dev/fattest.simplicity/src/componenttest/rules/repeater/EE9PackageReplacementHelper.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
+package componenttest.rules.repeater;
```
Comment (pnicolucci) [inline dev/fattest.simplicity/src/componenttest/rules/repeater/EE9PackageReplacementHelper.java:1]: We need a copyright header on this new class

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java (hunk 1 above)
Comment (volosied) [reply dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java:333]: All of the tests within this class do not use CDI, so I removed SkipForRepeat("CDI-2.0") from each method and added it to the class itself.

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#run-libby-bot

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java [hunk 7]
```diff
@@ -170,7 +168,7 @@ public void testEL30MethodNotFoundException() throws Exception {
         assertEquals("Expected " + 500 + " status code was not returned!",
                      500, response.getResponseCode());
 
-        assertTrue("The response did not contain: javax.el.MethodNotFoundException", response.getText().contains("javax.el.MethodNotFoundException"));
+        assertTrue("The response did not contain: javax.el.MethodNotFoundException", response.getText().contains("el.MethodNotFoundException"));
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java:171]: Can we do anything with the transformer here so we don't need to remove the javax? In addition we should clean up the assert messages if we're changing what we're actually looking for.

**Comment by volosied** [comment]
#build

**Comment by pnicolucci** [comment]
#run-libby-bot

**Pull request merged**